pinot-common/src/test/java/org/apache/pinot/common/utils/FALFInternerTest.java

```diff
@@ -0,0 +1,168 @@
+    Interner<String> falfInternerCustomHash = new FALFInterner(nUniqueObjs, s -> hashCode((String) s), Objects::equals);
+
+    // Go over all objects and intern them using both exact and FALF interners
+    int nHits1 = runInterning(allObjs, exactInterner, true);
+    int nHits2 = runInterning(allObjs, falfInterner, true);
+    int nHits3 = runInterning(allObjs, falfInternerCustomHash, true);
+
+    System.out.println(nHits1);
+    System.out.println(nHits2);
+    System.out.println(nHits3);
+
+    // For the exact interner, we should get a hit for each object except the
+    // first nUniqueObjs.
+    Assert.assertEquals(nHits1, nTotalObjs - nUniqueObjs);
+
+    // For the FALF interner, due to its fixed size and thus almost inevitable hash
+    // collisions, the number of hits is smaller. Let's verify that it's not too small, though.
+    Assert.assertTrue(nHits2 > (nTotalObjs - nUniqueObjs) * 0.4);
+
+    // With the better hash function, FALF interner should have more hits
+    Assert.assertTrue(nHits3 > (nTotalObjs - nUniqueObjs) * 0.6);
+
+    // Ad hoc benchmarking code. Disabled to avoid test slowdown.
+    // In one run the MacBook laptop, FALFInterner below performs nearly twice faster
+    // (1217 ms vs 2230 ms) With custom hash function, FALFInterner's speed is about the
+    // same as the Guava interner.
+//    for (int j = 0; j < 3; j++) {
+//      long time0 = System.currentTimeMillis();
+//      long totNHits = 0;
+//      for (int i = 0; i < 10000; i++) {
+//        totNHits += runInterning(allObjs, exactInterner, false);
+//      }
+//      long time1 = System.currentTimeMillis();
+//      System.out.println("Guava interner. totNHits = " + totNHits + ", time = " + (time1 - time0));
+//
+//      time0 = System.currentTimeMillis();
+//      totNHits = 0;
+//      for (int i = 0; i < 10000; i++) {
+//        totNHits += runInterning(allObjs, falfInterner, false);
+//      }
+//      time1 = System.currentTimeMillis();
+//      System.out.println("FALF interner. totNHits = " + totNHits + ", time = " + (time1 - time0));
+//
+//      time0 = System.currentTimeMillis();
+//      totNHits = 0;
+//      for (int i = 0; i < 10000; i++) {
+//        totNHits += runInterning(allObjs, falfInternerCustomHash, false);
+//      }
+//      time1 = System.currentTimeMillis();
+//      System.out.println("FALF interner Custom Hash. totNHits = " + totNHits + ", time = " + (time1 - time0));
+//    }
+  }
+
+  private int runInterning(String[] objs, Interner<String> interner, boolean performAssert) {
+    int nHits = 0;
+    for (String origObj : objs) {
+      String internedObj = interner.intern(origObj);
+      if (performAssert) {
+        Assert.assertEquals(origObj, internedObj);
+      }
+      if (origObj != internedObj) {
+        nHits++;
+      }
+    }
+    return nHits;
+  }
+
+  // Custom hash code implementation, that gives better distribution than standard hashCode()
```

Comment:
Same question. Why is all this goodness within a test class ? Seems like it can benefit regular code path as well